content/300-guides/050-database/830-prismas-type-system.mdx

```diff
@@ -0,0 +1,152 @@
+
+### Default type mappings
+
+To allow you to get started with our core scalar types, Prisma provides _default type mappings_ that map each scalar type to a default type in the underlying database. For example:
+
+- by default Prisma's `String` type gets mapped to PostgreSQL's `text` type and MySQL's `varchar` type
+- by default Prisma's `DateTime` type gets mapped to PostgreSQL's `timestamp(3)` type and SQL Server's `datetime2` type
+
+See Prisma's [database connector pages](/concepts/database-connectors) for the default type mappings for a given database. For example, [this table](/concepts/database-connectors/postgresql#prisma-migrate) gives the default type mappings for PostgreSQL. To see the default type mappings for a given database, see the [model field scalar types section]() of the Prisma schema reference. For example, [this table](/reference/api-reference/prisma-schema-reference#model-field-scalar-types) gives the default type mappings for the `String` scalar type.
+
+### Native type mappings
+
+Sometimes you may need to use a more specific database type that is not one of the default type mappings for your Prisma type. For this purpose, Prisma provides [native type attributes](/concepts/components/prisma-schema/data-model#native-types-mapping) to refine the core scalar types. For example, in the `createdAt` field of your `Post` model above you may want to use a date-only column in your underlying PostgreSQL database, by using the `date` type instead of the default type mapping of `timestamp(3)`. To do this, add a `@db.Date` native type attribute to the `createdAt` field:
+
+```prisma file=schema.prisma
+model Post {
+  id        Int      @id
+  title     String
+  createdAt DateTime @db.Date
+}
+```
+
+## How to introspect database types
+
+When you introspect your database, Prisma will take the database type of each table column and represent it in your Prisma schema using the correct Prisma type for the corresponding model field. If the database type is not the default database type for that Prisma scalar type, Prisma will also add a native type attribute.
+
+As an example, take a `User` table in a PostgreSQL database, with:
+
+- an `id` column with a data type of `serial`
+- a `name` column with a data type of `text`
+- an `isActive` column with a data type of `boolean`
+
+You can create this with the following SQL command:
+
+```sql
+CREATE TABLE "public"."User" (
+  id serial PRIMARY KEY NOT NULL,
+  name text NOT NULL,
+  "isActive" boolean NOT NULL
+);
+```
+
+Introspect your database with the following command run from the root directory of your project:
```

Comment:
This skips over the bit how you need a `datasource` in your schema, and a connection string defined somewhere, often as an env var that is used in the schema. Might be fine to skip over that - maybe we have a good place to link to though?

Reply:
I've added a link to the page on introspection further up. Not sure we need to go into more detail here, I think most of the docs implicitly assume you've worked out the connecting to your database part